common/lib/ktypes.ts

```diff
@@ -60,6 +60,38 @@ type CollectionSupports = {
   IOpenNFTsV3?: boolean;
 };
 
+///////////////////////////////////////////////////
+// Exclusives storage parameters for Ipfs | Swarm
+///////////////////////////////////////////////////
+type Without<T> = { [P in keyof T]?: undefined };
+type XOR<T, U> = (Without<T> & U) | (Without<U> & T);
+
+////////
+type ipfsImgType = {
+  ipfs?: string;
+};
+
+type swarmImgType = {
+  swarm?: string;
+};
+
+////////
+type ipfsType = {
+  ipfs?: string;
+  ipfsJson?: string;
+};
+
+type swarmType = {
+  swarm?: string;
+  swarmJson?: string;
+};
+
+////////
+type storageType = XOR<ipfsType, swarmType>;
+
+type storageImgType = XOR<ipfsImgType, swarmImgType>;
```

Comment:
Not necessary : you can use ipfsType , swarmType instead

Reply:
do you mean using ipfsType and swarmType for type NftMetaData ?
even if metadata type never use Json parameter?